app/controllers/email_controller.rb

```diff
@@ -6,120 +6,61 @@ class EmailController < ApplicationController
   skip_before_action :check_xhr, :preload_json, :redirect_to_login_if_required
 
   def unsubscribe
-    @not_found = true
+    @digest_unsubscribe = false
     @watched_count = nil
 
-    if key = UnsubscribeKey.find_by(key: params[:key])
-      if @user = key.user
-        post = key.post
-        @topic = post&.topic || key.topic
-        @digest_unsubscribe = !@topic && !SiteSetting.disable_digest_emails
-        @type = key.unsubscribe_key_type
-        @not_found = false
-
-        if current_user.present? && (@user != current_user)
-          @different_user = @user.name
-          @return_url = request.original_url
-        end
-
-        watching = TopicUser.notification_levels[:watching]
-
-        @unsubscribed_from_all = @user.user_option.unsubscribed_from_all?
-
-        if @topic
-          @watching_topic = TopicUser.exists?(user_id: @user.id, notification_level: watching, topic_id: @topic.id)
-          if @topic.category_id
-            if CategoryUser.exists?(user_id: @user.id, notification_level: CategoryUser.watching_levels, category_id: @topic.category_id)
-              @watched_count = TopicUser.joins(:topic)
-                .where(user: @user, notification_level: watching, "topics.category_id" => @topic.category_id)
-                .count
-            end
-          end
-        else
-          @digest_frequencies = digest_frequencies(@user)
-        end
+    key = UnsubscribeKey.find_by(key: params[:key])
+    @not_found = key.nil?
+
+    if !@not_found
```

Comment:
I know it existed like this already, but I'd prefer `@found` variable 🙂 (instead of "not not found" etc. 😉)